<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Tutorial: Clonal Samples (Consensus Mode) &mdash; breseq 0.33.2 documentation</title>
    
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/custom.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.33.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="breseq 0.33.2 documentation" href="index.html" />
    <link rel="next" title="Tutorial: Population Samples (Polymorphism Mode)" href="tutorial_populations.html" />
    <link rel="prev" title="Acknowledgments" href="acknowledgments.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="tutorial_populations.html" title="Tutorial: Population Samples (Polymorphism Mode)"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="acknowledgments.html" title="Acknowledgments"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">breseq 0.33.2 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="tutorial-clonal-samples-consensus-mode">
<span id="tutorial-clones"></span><h1>Tutorial: Clonal Samples (Consensus Mode)<a class="headerlink" href="#tutorial-clonal-samples-consensus-mode" title="Permalink to this headline">¶</a></h1>
<p>This tutorial expands on the <a class="reference internal" href="test_drive.html#test-drive"><span>Test Drive</span></a>. You will analyze mutations in the genomes of multiple clones isolated from population Ara-3 of the Lenski long-term evolution experiment (LTEE). A complex mutation is present in these samples that was necessary for evolution of the aerobic citrate utilization trait (Cit+). In addition to some tips on <strong class="program">breseq</strong> usage and examples of interpreting more complex mutations in the output, this tutorial also introduces functionality in the <strong class="program">gdtools</strong> utility command that can be used to compare and analyze mutations in an entire set of evolved genomes.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This tutorial was created for the EMBO Practical Course <a class="reference external" href="http://events.embo.org/15-htp-sequencing/">Measuring intra-species diversity using high-throughput sequencing</a> held 27–31 July 2015 in Oeiras, Portugal.</p>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">If you encounter any <strong class="program">breseq</strong> or <strong class="program">gdtools</strong> errors or crashes in running this tutorial, please <a class="reference external" href="https://github.com/barricklab/breseq/issues">report issues on GitHub</a>.</p>
</div>
<div class="section" id="download-data-files">
<h2>1. Download data files<a class="headerlink" href="#download-data-files" title="Permalink to this headline">¶</a></h2>
<p>First, create a directory called <code class="docutils literal"><span class="pre">tutorial_clonal</span></code>:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>mkdir tutorial_clones
<span class="nv">$ </span><span class="nb">cd </span>tutorial_clones
</pre></div>
</div>
<div class="section" id="reference-sequence">
<h3>Reference sequence<a class="headerlink" href="#reference-sequence" title="Permalink to this headline">¶</a></h3>
<p><strong class="program">breseq</strong> prefers the reference sequence in <a class="reference external" href="http://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html">Genbank</a> or <a class="reference external" href="http://gmod.org/wiki/GFF">GFF3</a> format. In this example, the reference sequence is <em>Escherichia coli</em> B strain REL606. The Genbank (Refseq) accession number is: <strong>NC_012967</strong>. For this tutorial, we are going to use an updated version of this file that contains richer gene annotation information than the version available from NCBI.</p>
<ul class="simple">
<li><a class="reference external" href="http://barricklab.org/release/breseq_tutorial/REL606.gbk.gz">Download REL606.gbk via this link</a></li>
</ul>
</div>
<div class="section" id="read-files">
<h3>Read files<a class="headerlink" href="#read-files" title="Permalink to this headline">¶</a></h3>
<p>In this tutorial, we&#8217;re going to use Illumina genome re-sequencing data from <em>E. coli</em> strains that evolved for up to 40,000 generations in population Ara-3 from the Lenski long-term evolution experiment <a class="reference internal" href="references.html#blount2011" id="id1">[Blount2011]</a>. This data is available in the European Nucleotide Archive (ENA). Go to <a class="reference external" href="http://www.ebi.ac.uk/">http://www.ebi.ac.uk/</a> and search for one of the genomes in the table below (or click on the accession number to be taken directly to the download page for that clone).</p>
<table border="1" class="docutils">
<colgroup>
<col width="14%" />
<col width="14%" />
<col width="71%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Accession</th>
<th class="head">Clone</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><a class="reference external" href="http://www.ebi.ac.uk/ena/data/view/SRR098289">SRR098289</a></td>
<td>ZDB564</td>
<td>Cit+ clone isolated at generation 31,500</td>
</tr>
<tr class="row-odd"><td><a class="reference external" href="http://www.ebi.ac.uk/ena/data/view/SRR098042">SRR098042</a></td>
<td>ZDB172</td>
<td>Cit+ clone isolated at generation 32,000</td>
</tr>
<tr class="row-even"><td><a class="reference external" href="http://www.ebi.ac.uk/ena/data/view/SRR098040">SRR098040</a></td>
<td>ZDB143</td>
<td>Cit+ clone isolated at generation 32,500</td>
</tr>
<tr class="row-odd"><td><a class="reference external" href="http://www.ebi.ac.uk/ena/data/view/SRR097977">SRR097977</a></td>
<td>CZB152</td>
<td>Cit+ clone isolated at generation 33,000</td>
</tr>
<tr class="row-even"><td><a class="reference external" href="http://www.ebi.ac.uk/ena/data/view/SRR098026">SRR098026</a></td>
<td>CZB154</td>
<td>Cit+ clone isolated at generation 33,000</td>
</tr>
<tr class="row-odd"><td><a class="reference external" href="http://www.ebi.ac.uk/ena/data/view/SRR098034">SRR098034</a></td>
<td>ZDB83</td>
<td>Cit+ clone isolated at generation 34,000</td>
</tr>
</tbody>
</table>
<p>Download the FASTQ files for your chosen sample into your <strong>tutorial_clones</strong> directory.</p>
<p>These particular clonal isolates were chosen to illustrate particular aspects of interpreting <cite>breseq</cite> output. If you would like to investigate data corresponding to any of the 29 sequenced genomes from this population described in <a class="reference internal" href="references.html#blount2011" id="id2">[Blount2011]</a> check out <a class="reference external" href="http://www.ebi.ac.uk/ena/data/view/SRP004752">SRP004752</a>.</p>
</div>
</div>
<div class="section" id="run-breseq">
<h2>2. Run <strong class="program">breseq</strong><a class="headerlink" href="#run-breseq" title="Permalink to this headline">¶</a></h2>
<p>Check to be sure that you have changed into the <strong>tutorial_clones</strong> directory, that you have all of the input files, and that you have decompressed these files.</p>
<p>Now, run breseq using a command of this form (replacing strain numbers and FASTQ file names as appropriate):</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>breseq -l <span class="m">60</span> -j <span class="m">8</span> -o ZDB83_output -r REL606.gbk SRR098034.fastq
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">We are using the <strong>-l 60</strong> option to limit the amount of read data that we use to a nominal coverage of 60-fold to speed up this <strong class="program">breseq</strong> run. Generally, setting this option to somewhere in the range of 80 to 120 is recommended. In our experience, using more read data than that is usually not helpful for analyzing clonal samples. It just slows down the analysis!</p>
</div>
<p>What is the <strong>-j</strong> option doing?</p>
<div class="toggle container">
<div class="header container">
<strong>Click on the triangle to reveal the answer</strong></div>
<div class="text container">
You can show the most commonly used <strong class="program">breseq</strong> options by calling it with no arguments, and you can show all options by adding the <strong>-h</strong> flag.</div>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>breseq
<span class="nv">$ </span>breseq -h
</pre></div>
</div>
</div>
</div>
<div class="section" id="open-breseq-output">
<h2>3. Open <strong class="program">breseq</strong> output<a class="headerlink" href="#open-breseq-output" title="Permalink to this headline">¶</a></h2>
<p>Using a web browser, open the file <code class="docutils literal"><span class="pre">index.html</span></code> in the output directory. This file contains tables describing the predicted mutations and also evidence that may indicate that there are other mutations that <strong class="program">breseq</strong> could not resolve into precise genetic changes that you should further examine. The tables in this HTML file are more fully described in the <a class="reference internal" href="output.html#output-format"><span>Output</span></a> section of the <strong class="program">breseq</strong> manual.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If you run into problems generating the <strong class="program">breseq</strong> output and would like to continue this tutorial, please <a class="reference external" href="http://barricklab.org/release/breseq_tutorial/ZDB83_output.tgz">download pre-generated results for ZDB83</a> (created using <cite>breseq</cite> version 0.26.1).</p>
</div>
<p>Now, browse the upper table in the results of <strong>Mutation predictions</strong>. Click on <strong>RA</strong>, <strong>JC</strong>, and <strong>MC</strong> links to view the evidence <cite>breseq</cite> used to predict these mutations. There should be a variety of base substitutions, indels, large deletions, and transposon (IS element) insertions in your results.</p>
<p>Consider these questions:</p>
<ol class="arabic simple">
<li>Click on the <strong>MC</strong> link on the line for a large deletion that is mediated by an IS element or between IS elements (look in the <strong>annotation</strong> column). What are the red and blue lines on the coverage graphs?</li>
</ol>
<div class="toggle container">
<div class="header container">
<strong>Click for an example of an IS-mediated MC evidence item</strong></div>
<div class="text container">
<div class="figure">
<img alt="_images/ZDB83.waaT.IS-mediated-deletion.png" src="_images/ZDB83.waaT.IS-mediated-deletion.png" />
</div>
</div>
</div>
<div class="toggle container">
<div class="header container">
<strong>Click to reveal the answers</strong></div>
<div class="text container">
Blue lines are coverage from uniquely mapped reads.  Red lines indicate coverage from reads that map equally well to genomic repeat regions that also exist elsewhere in the genome. You should see red coverage on one or both sides of the deletion that correspond to the  IS elements because these transposons occur in mutliple identical copies around the genome. The read length is too short to tell which copy a read came from if it maps to the middle of these elements. They are hich have lengths of around 1000-1500 base pairs).</div>
</div>
<p>Next, scroll down to the <strong>Unnassigned...</strong> tables that are near the bottom of the page. Click on a few of these evidence items and examine the read coverage depth or alignment of mapped reads. Can you tell which <strong>MC</strong> items are associated with <strong>JC</strong> items and what mutations may have happened in the evolved genome? This can take practice, so we&#8217;ll work a few examples.</p>
<ol class="arabic simple" start="2">
<li>Find an <strong>Unassigned missing coverage evidence</strong> entry in an <em>rhs</em> or <em>rrl</em> gene. Examine the read coverage graph. What is the most likely mutation here?</li>
</ol>
<div class="toggle container">
<div class="header container">
<strong>Click for an example rrlA MC image</strong></div>
<div class="text container">
<div class="figure">
<img alt="_images/ZDB83.rrlA.MC.png" src="_images/ZDB83.rrlA.MC.png" />
</div>
</div>
</div>
<div class="toggle container">
<div class="header container">
<strong>Click to reveal possible answers</strong></div>
<div class="text container">
These variants apparently missing small segments of uniquely mapped reads in a mostly repetitive region are most likely explained by non-allelic gene conversions. Recombination converts one copy of the near-repeat to have the exacts sequence of another copy of the near-repeat. Since we are mapping reads to the reference genome, this makes it appear that the sequence here was deleted, when it was just changed to mirror the other copy exactly. If you looked at the coverage of the other copy, then you would likely see correspondingly higher coverage at any unique segments.</div>
</div>
<ol class="arabic simple" start="3">
<li>Can you find a situation where an observed <strong>Unassigned missing coverage evidence</strong> item also has junctions to new IS elements on each of its boundaries? What mutation event or events could explain this genetic variant?</li>
</ol>
<div class="toggle container">
<div class="header container">
<strong>Click for an example of an unassigned MC image of this kind</strong></div>
<div class="text container">
<div class="figure">
<img alt="_images/ZDB83.ECB02816-ECB02836.MC.png" src="_images/ZDB83.ECB02816-ECB02836.MC.png" />
</div>
</div>
</div>
<div class="toggle container">
<div class="header container">
<strong>Click to reveal possible answers</strong></div>
<div class="text container">
<p>In general, if you see sharp drops to zero of blue (unique) coverage on both margins of the deletion, then there will be two junctions to a transposable IS element, one for each side of the deletion. Thus, in the sequenced genome there is one copy of the IS element bridging between the ends of the deleted region. The two mutations that led to this final genome are likely first an insertion of one copy of the IS element, then an insertion of another IS element copy in the same orientation, followed by a deletion between them that also removes one IS copy by homologous recombination between; or, alternatively, an IS mediated deletion adjacent to the new IS copy. By examining other clones in the population, it may sometimes be possible to find out the location of the first IS element insertion if it is ever observed separately before the deletion.</p>
<p>In this case, these two JC evidence items in the unassigned table are the ones that correspond to two sides of a transposable element (IS*150* copy).</p>
<div class="figure">
<img alt="_images/ZDB83.ECB02816-ECB02836.JC.png" src="_images/ZDB83.ECB02816-ECB02836.JC.png" />
</div>
</div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If you&#8217;d like to explore the mapped reads in more detail than is possible by looking at the alignments and coverage graphs that <strong class="program">breseq</strong> automatically generates, read the section of the manual on <a class="reference internal" href="output.html#viewing-in-igv"><span>Viewing Output / Aligned Reads in the IGV</span></a>.</p>
</div>
</div>
<div class="section" id="resolving-the-cit-mutation">
<h2>4. Resolving the Cit+ mutation<a class="headerlink" href="#resolving-the-cit-mutation" title="Permalink to this headline">¶</a></h2>
<p>The mutation that caused the Cit+ phenotype encompasses the <em>citT</em> gene (which encodes a citrate:succinate antiporter). It is detected as an <strong>unassigned JC</strong> evidence item in the <strong class="program">breseq</strong> results (found in a table lower on the results page). It also leads to a change in coverage in this region of the genome. Your task now is to figure out what mutation or mutations gave rise to the evolved genome architecture.</p>
<p>The steps that can be used to find the answer are illustrated using ZDB83 as an example, but details of the evolved <em>citT</em> arrangement are different in each of the example genomes. Be sure to pay attention to what is actually happening in your results!</p>
<div class="section" id="a-rnk-citg-junction">
<h3>A. <em>rnk-citG</em> junction<a class="headerlink" href="#a-rnk-citg-junction" title="Permalink to this headline">¶</a></h3>
<p>Find the junction annotated as <em>rnk-citG</em> in the <strong>Unassigned Junction</strong> table. It should look something like this.</p>
<div class="figure">
<img alt="_images/ZDB83.rnk-citG.JC.png" src="_images/ZDB83.rnk-citG.JC.png" />
</div>
<p>What does this junction mean? What regions of the genome are juxtaposed in a way that they were not in the reference genome?</p>
<div class="toggle container">
<div class="header container">
<strong>Click to see the answer</strong></div>
<div class="container">
The junction means that reads connect position 625889 (extending to higher coordinates) to position 628821 (extending to lower coordinates). Don&#8217;t get what this means? Click on the * link and it will show the two snippets of reference genome and how they are joined by split-read alignments across the junction. The image below also shows what the new arrangement would look like. It&#8217;s evidence for a tandem head-to-tail amplification.</div>
<div class="figure">
<a class="reference internal image-reference" href="_images/ZDB83.rnk-citG.JC.schematic.png"><img alt="_images/ZDB83.rnk-citG.JC.schematic.png" src="_images/ZDB83.rnk-citG.JC.schematic.png" style="width: 800px;" /></a>
</div>
</div>
</div>
<div class="section" id="b-zoomed-in-coverage">
<h3>B. Zoomed-in coverage<a class="headerlink" href="#b-zoomed-in-coverage" title="Permalink to this headline">¶</a></h3>
<p>If the region is duplicated (or amplified to even more copies) then there should also be an increase in the genomic coverage of this region. Use the <strong>breseq BAM2COV</strong> subcommand to create a zoomed-in view of this region to try to better discern its boundaries and copy number. Alternatively, you can navigate to this region of the genome by loading the <strong class="program">breseq</strong> output in IGV (as described above).</p>
<p>You can get the help by running this command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>breseq BAM2COV -h
</pre></div>
</div>
<p>Here&#8217;s an example of applying <strong>BAM2COV</strong> to one of these genomes (In this case, by running it from inside the <strong class="program">breseq</strong> output directory, we can use a simpler version of the command that leaves off some options and has extra information available about the average coverage across the genome):</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span><span class="nb">cd </span>ZDB83_output
<span class="nv">$ </span>breseq BAM2COV -a REL606:624500-630500
</pre></div>
</div>
<p>Here&#8217;s an example of the corresponding output:</p>
<div class="figure">
<a class="reference internal image-reference" href="_images/ZDB83.rnk-citG.zoom-in.png"><img alt="_images/ZDB83.rnk-citG.zoom-in.png" src="_images/ZDB83.rnk-citG.zoom-in.png" style="width: 792px;" /></a>
</div>
</div>
<div class="section" id="c-add-the-amplification-to-the-genomediff-file">
<h3>C. Add the amplification to the <cite>GenomeDiff</cite> file<a class="headerlink" href="#c-add-the-amplification-to-the-genomediff-file" title="Permalink to this headline">¶</a></h3>
<p><strong class="program">GenomeDiff</strong> files are text files output by <strong class="program">breseq</strong> that describe all of the mutations detected in a sample. (They&#8217;re like &#8220;diff&#8221; files for code, describing just the small number of changes needed to &#8220;patch&#8221; a text file to create an updated version.) <strong class="program">GenomeDiff</strong> files contain similar information to VCF files describing genetic variants, but they are more focused on describing mutational events. They make it possible to (1) easily annotate structural variants such as transposon insertions and amplifications, (2) include gene annotation information, and (3) set the order with which to apply mutations, which is sometimes necessary to fully describe evolutionary data. The <strong class="program">gdtools</strong> command exists to perform various functions on <strong class="program">GenomeDiff</strong> files, as we&#8217;ll see in the next few sections (much like the <strong>vcftools</strong> command) .</p>
<p>Since <strong class="program">breseq</strong> could not resolve the <em>rnk-citG</em> amplification, this very important mutation is not present in the <strong class="program">GenomeDiff</strong> output file, so it won&#8217;t be included in downstream analysis steps unless you edit this file and add a description of the event to the output.</p>
<p>To illustrate how to do this, make a copy of the file <code class="docutils literal"><span class="pre">output/output.gd</span></code> from the <strong class="program">breseq</strong> output directory for your run. Name it something like <code class="docutils literal"><span class="pre">ZDB83.gd</span></code> and keep it outside of the <strong class="program">breseq</strong> output folder. If you change your working directory back to <code class="docutils literal"><span class="pre">tutorial_clones</span></code>, then a command like this will manage that:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>cp ZDB83_output/output/output.gd ZDB83.gd
</pre></div>
</div>
<p>Open this file in a text editor to see how it represents mutations and other information that is in the HTML output files in a parsable format. Refer to <a class="reference internal" href="gd_format.html#genomediff-usage"><span>GenomeDiff Format</span></a> for a full description of the <strong class="program">GenomeDiff</strong> appendix of the <strong class="program">breseq</strong> manual for a full description of the file format.</p>
<p>Let&#8217;s add an <strong>AMP</strong> line describing the <em>rnk-citG</em> amplification to our new <code class="docutils literal"><span class="pre">ZDB83.gd</span></code> file. You can add this line anywhere in the file.</p>
<div class="toggle container">
<div class="header container">
<strong>Try to do this yourself. Here&#8217;s a hint to get you started</strong></div>
<div class="container">
The line should start this way:</div>
<div class="highlight-text"><div class="highlight"><pre>AMP       .       .       REL606  &lt;insert additional columns here!&gt;
</pre></div>
</div>
<p>The <em>1st column</em> defines the type of mutation as an <strong>AMP</strong>. The <em>2nd and 3rd columns</em> are IDs that don&#8217;t have to be set for manually added mutations. They enable one to link mutations and evidence items that support them. Using a dot or period (<strong>.</strong>) just signifies that you are leaving them blank. The <em>4th column</em> is the <strong>seq_id</strong> identifying the reference sequence fragment on which the mutation is located. There&#8217;s only one choice for this <em>E. coli</em> with a single chromosome: <strong>REL606</strong>.</p>
</div>
<div class="toggle container">
<div class="header container">
<strong>Here&#8217;s the answer for ZDB83</strong></div>
<div class="highlight-text"><div class="highlight"><pre>AMP       .       .       REL606  625889  2933    4
</pre></div>
</div>
<div class="container">
The <em>5th column</em> is the starting coordinate of the amplification (straight from the <strong>JC</strong> evidence). The <em>6th column</em> is the size of the amplification (calculated as 628821 - 625889 + 1 = 2933 in this case). The <em>7th column</em> is the new copy number of this region in the mutated genome (4 copies in this case).</div>
</div>
<p>You can run <strong>gdtools VALIDATE</strong> to check your syntax for errors. For example, using the command:</p>
<div class="highlight-bash"><div class="highlight"><pre>gdtools VALIDATE -r REL606.gbk ZDB83.gd
</pre></div>
</div>
</div>
</div>
<div class="section" id="generating-a-mutated-reference-sequence">
<h2>5. Generating a mutated reference sequence<a class="headerlink" href="#generating-a-mutated-reference-sequence" title="Permalink to this headline">¶</a></h2>
<p>If you manually edit <strong class="program">GenomeDiff</strong> files to resolve evidence into new mutations, it&#8217;s best to check that you&#8217;ve described the mutations correctly in your curated file (with the right start positions, size, variant bases, etc). You can do this by simulating the mutant genome that you have described in the <strong class="program">GenomeDiff</strong> file and then re-querying the sequencing read data against it in a second <strong class="program">breseq</strong> pass. Ideally, no variants will be detected when you do this if your list of mutations is complete! It&#8217;s also possible that mapping to this mutated reference genome will enable you to detect new variants – because reads that did not match the original genome sufficiently well now map to the mutated reference sequence.</p>
<p>Use a command like this to generate the mutant genome sequence:</p>
<div class="highlight-bash"><div class="highlight"><pre>gdtools APPLY -f GFF3 -o ZDB83.gff3 -r REL606.gbk ZDB83.gd
</pre></div>
</div>
<p>Now start a new <cite>breseq</cite> run using a command like this:</p>
<div class="highlight-bash"><div class="highlight"><pre>breseq -j <span class="m">4</span> -l <span class="m">60</span> -o ZDB83_mutated_output -r ZDB83.gff3 SRR098034.fastq
</pre></div>
</div>
<p>Wait for this to run and then examine the output (you probably want to continue this tutorial in another window while you wait for that command to complete).</p>
<p>You should see the all of the previously predicted mutations and the <em>rnk-citG</em> junction disappear in the new output! If you make a coverage plot of the <em>citT</em> gene and the surrounding region, the coverage will also now be in red, indicating that reads no longer uniquely map to it because there are now multiple copies.</p>
<p>Note that coordinates of genes are shifted in the mutated reference genome due to mutations! If you want to find out where the <em>citT</em> gene copies are located to use <strong>gdtools BAM2COV</strong>, you can use this command to show the relevant lines of the GFF3 reference file:</p>
<div class="highlight-bash"><div class="highlight"><pre>grep citT ZDB83.gff3
</pre></div>
</div>
</div>
<div class="section" id="characterizing-genetic-diversity-and-genome-evolution">
<h2>6. Characterizing genetic diversity and genome evolution<a class="headerlink" href="#characterizing-genetic-diversity-and-genome-evolution" title="Permalink to this headline">¶</a></h2>
<p>At this point in the tutorial, you can branch off and explore multiple topics related to understanding genome evolution in this <em>E. coli</em> population. Normally, you would generate the &#8220;raw&#8221; <strong class="program">breseq</strong> output for all of the genomes in this population and curate these <strong class="program">GenomeDiff</strong> files to resolve unassigned evidence. This can be a very time-consuming process (the <strong class="program">breseq</strong> runs themselves, but especially getting all of the details correct for resolving the unassigned evidence).</p>
<p>So, to get the most out of this tutorial in the shortest amount of time, please continue by <a class="reference external" href="http://barricklab.org/release/breseq_tutorial/curated_gd.tgz">downloading curated GenomeDiff files</a> for the 29 clones from this population described in <a class="reference internal" href="references.html#blount2011" id="id3">[Blount2011]</a>.</p>
<p>Place this archive inside of your <code class="docutils literal"><span class="pre">tutorial_clones</span></code> directory. Decompress and change into the resulting directory, which is full of the curated <strong class="program">GenomeDiff</strong> files.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>tar -xvzf curated_gd.tgz
<span class="nv">$ </span><span class="nb">cd </span>curated_gd
</pre></div>
</div>
<p>Note that these files have been edited in two ways from the raw <cite>breseq</cite> output. First, unassigned evidence has been resolved into mutations by manually editing the <cite>GenomeDiff</cite> files. Second, additional metadata has been added in the header sections. This data can be used by the <strong class="program">gdtools</strong> utilities that will be demonstrated in these examples.</p>
<p>For example, the header of the file <code class="docutils literal"><span class="pre">Ara-3_34000gen_ZDB83.gd</span></code> looks like this:</p>
<div class="highlight-text"><div class="highlight"><pre>#=GENOME_DIFF 1.0
#=TIME       34000
#=POPULATION Ara-3
#=TREATMENT  LTEE
#=CLONE      ZDB83
</pre></div>
</div>
<div class="section" id="example-1-compare-mutations-in-different-genomes">
<h3>Example 1. Compare mutations in different genomes<a class="headerlink" href="#example-1-compare-mutations-in-different-genomes" title="Permalink to this headline">¶</a></h3>
<p>You can generate a table that enables you to compare the presence/absence of the mutations in multiple genomes using the <strong>gdtools COMPARE</strong> command. Let&#8217;s generate a comparison table including the 4 the genomes from generations 15,000 and earlier. Try to figure out how to do this from the help:</p>
<div class="highlight-bash"><div class="highlight"><pre>gdtools COMPARE -h
</pre></div>
</div>
<div class="toggle container">
<div class="header container">
<strong>Here&#8217;s an example command</strong></div>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>gdtools COMPARE -o ../compare.html -r ../REL606.gbk Ara-3_2000gen_REL1166A.gd Ara-3_5000gen_ZDB409.gd Ara-3_10000gen_ZDB429.gd Ara-3_15000gen_ZDB446.gd
</pre></div>
</div>
</div>
<p>Answer these questions from the <strong>gdtools COMPARE</strong> output:</p>
<ol class="arabic simple">
<li>In what order did mutations in these four genes occur in the population: <em>fis</em>, <em>hslU</em>, <em>mrdB</em>, <em>ybaL</em>?</li>
</ol>
<div class="toggle container">
<div class="header container">
<strong>Show the answer</strong></div>
<div class="text container">
<em>mrdB</em> (2000+ generations) &lt; <em>hslU</em> and <em>fis</em> (5000+ generations) &lt; <em>ybaL</em> (10000+ generations)</div>
</div>
<ol class="arabic simple" start="2">
<li>Do you see any mutations that are off the main line of descent? (That is, they only appear in one or more early clones from a lineage that appears to have later gone extinct, like from 2,000 or 5,000 generations.)</li>
</ol>
<div class="toggle container">
<div class="header container">
<strong>Show me some examples</strong></div>
<div class="text container">
For example, mutations in the genes <em>eaeH</em>, <em>infB</em>, and <em>fadL</em>.</div>
</div>
</div>
<div class="section" id="example-2-analyze-rates-and-nature-of-genome-evolution">
<h3>Example 2. Analyze rates and nature of genome evolution<a class="headerlink" href="#example-2-analyze-rates-and-nature-of-genome-evolution" title="Permalink to this headline">¶</a></h3>
<p>You can count the number of mutations in each genome using the <strong>gdtools COUNT</strong> command. Try to figure out how to run it from the help:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>gdtools COUNT -h
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Save yourself some typing by putting <code class="docutils literal"><span class="pre">`ls</span> <span class="pre">*.gd`</span></code> in your command line for the <strong class="program">GenomeDiff</strong> arguments! The backticks run the enclosed command and insert the output into your command line. This requires that you run the command from inside the <code class="docutils literal"><span class="pre">curated_gd</span></code> folder.</p>
</div>
<div class="toggle container">
<div class="header container">
<strong>Here&#8217;s an example command</strong></div>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>gdtools COUNT -o ../count.csv -r ../REL606.gbk <span class="sb">`</span>ls *.gd<span class="sb">`</span>
</pre></div>
</div>
</div>
<p>Open the comma-separated values (CSV) output file <code class="docutils literal"><span class="pre">count.csv</span></code> in a <strong>spreadsheet program</strong> or in <strong>R</strong>.</p>
<p>Answer these questions by loading the <strong>gdtools COMPARE</strong> output into <strong>R</strong>. It&#8217;s easiest if you change back into your main <code class="docutils literal"><span class="pre">tutorial_clones</span></code> directory first.</p>
<div class="highlight-R"><div class="highlight"><pre><span class="o">&gt;</span> count <span class="o">=</span> read.csv<span class="p">(</span><span class="s">&quot;count.csv&quot;</span><span class="p">,</span> header<span class="o">=</span><span class="bp">T</span><span class="p">)</span>
<span class="o">&gt;</span> <span class="kp">head</span><span class="p">(</span>count<span class="p">)</span>
</pre></div>
</div>
<ol class="arabic simple">
<li>Is the rate at which mutations accumulate in these genomes constant over time in this population? Graph the total number of mutations in each genome (<strong>total</strong> column) versus the generation when it was isolated (<strong>time</strong> column).</li>
</ol>
<div class="toggle container">
<div class="header container">
<strong>Show R commands</strong></div>
<div class="highlight-R"><div class="highlight"><pre><span class="o">&gt;</span> plot<span class="p">(</span>total <span class="o">~</span> time<span class="p">,</span> count<span class="p">)</span>
</pre></div>
</div>
<div class="toggle container">
<div class="header container">
Here&#8217;s the key thing to notice in the graph</div>
<div class="text container">
Some of the genomes from 36,000+ generations appear to have evolved an elevated mutation rate! They fall way above a line fit to the rest of the points.</div>
</div>
</div>
<ol class="arabic simple" start="2">
<li>There are 52 mutations in ZDB464, a clone isolated at 20,000 generations. If we did a separate evolution experiment under conditions of relaxed selection so that we were observing the spontaneous rate of mutations (i.e., a mutation accumulation experiment) and observed 47 mutations over 50,000 generations of evolution, could we conclude that the rate of genome evolution along the lineages leading to ZDB64 was faster or slower than the spontaneous mutation rate? What is the P-value for rejecting the null hypothesis of the same rate?</li>
</ol>
<div class="toggle container">
<div class="header container">
<strong>Show a hint</strong></div>
<div class="text container">
This is count data, so you&#8217;ll want to use a Poisson test. That&#8217;s the <code class="docutils literal"><span class="pre">poisson.test</span></code> function in <strong>R</strong>.</div>
</div>
<div class="toggle container">
<div class="header container">
<strong>Show the R commands</strong></div>
<div class="highlight-R"><div class="highlight"><pre><span class="o">&gt;</span> poisson.test<span class="p">(</span> <span class="kt">c</span><span class="p">(</span><span class="m">52</span><span class="p">,</span> <span class="m">47</span><span class="p">),</span> <span class="kt">c</span><span class="p">(</span><span class="m">20000</span><span class="p">,</span> <span class="m">50000</span><span class="p">))</span>

Comparison of Poisson rates

data<span class="o">:</span>  <span class="kt">c</span><span class="p">(</span><span class="m">52</span><span class="p">,</span> <span class="m">64</span><span class="p">)</span> time base<span class="o">:</span> <span class="kt">c</span><span class="p">(</span><span class="m">20000</span><span class="p">,</span> <span class="m">50000</span><span class="p">)</span>
count1 <span class="o">=</span> <span class="m">52</span><span class="p">,</span> expected count1 <span class="o">=</span> <span class="m">33.143</span><span class="p">,</span> p<span class="o">-</span>value <span class="o">=</span> <span class="m">0.0001932</span>
alternative hypothesis<span class="o">:</span> true rate ratio is not equal to <span class="m">1</span>
<span class="m">95</span> percent confidence interval<span class="o">:</span>
 <span class="m">1.381119</span> <span class="m">2.975263</span>
sample estimates<span class="o">:</span>
rate ratio
  <span class="m">2.03125</span>
</pre></div>
</div>
<div class="text container">
So, yes the rates differ significantly (P = 0.00019).</div>
</div>
</div>
<div class="section" id="example-3-analyze-unique-mutations-for-evidence-of-bias-and-selection">
<h3>Example 3. Analyze unique mutations for evidence of bias and selection<a class="headerlink" href="#example-3-analyze-unique-mutations-for-evidence-of-bias-and-selection" title="Permalink to this headline">¶</a></h3>
<p>Often, we only want to examine unique mutations within a population. Many of the mutations observed in the 5,000 generation clone are also found in the 20,000 generation clone, so just adding up the counts of mutations in each genome and then analyzing them for bias or selection would have a problem with pseudoreplication. Said another way, we don&#8217;t want to count these mutations, which most likely only happened once in the population, as two independent observation supporting a hypothesis. One mutational event should count as one observation, no matter how many genomes it appears in.</p>
<p>To define the set of unique mutations, we can use the <strong>gdtools UNION</strong> command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>gdtools UNION -h
</pre></div>
</div>
<p>You&#8217;ll want to exclude those outlier genomes for this analysis (the ones which have evolved a high mutation rate). It&#8217;s probably easiest to move or delete these three genomes from the <code class="docutils literal"><span class="pre">curated_gd</span></code> folder: <code class="docutils literal"><span class="pre">Ara-3_36000gen_ZDB96.gd</span></code>, <code class="docutils literal"><span class="pre">Ara-3_38000gen_ZDB107.gd</span></code>, and <code class="docutils literal"><span class="pre">Ara-3_40000gen_REL10979.gd</span></code>.</p>
<div class="toggle container">
<div class="header container">
<strong>Show the gdtools command</strong></div>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span><span class="nb">cd </span>curated_gd
<span class="nv">$ </span>gdtools UNION -o ../unique.gd <span class="sb">`</span>ls *.gd<span class="sb">`</span>
</pre></div>
</div>
</div>
<p>This time, run the <strong>gdtools COUNT</strong> on just the one resulting <code class="docutils literal"><span class="pre">unique.gd</span></code> file. <strong>Important:</strong> Include the <strong>-b</strong> option. This provides additional output of the total number of synonymous and nonsynonymous mutations possible in the genome.</p>
<div class="toggle container">
<div class="header container">
<strong>Show the answer</strong></div>
<div class="highlight-R"><div class="highlight"><pre><span class="o">$</span> cd <span class="m">..</span>
<span class="o">$</span> gdtools COUNT <span class="o">-</span>b <span class="o">-</span>o unique.csv <span class="o">-</span>r REL606.gbk unique.gd
</pre></div>
</div>
</div>
<p>Now, load <code class="docutils literal"><span class="pre">unique.csv</span></code> into <strong>R</strong> to answer these questions:</p>
<ol class="arabic simple">
<li>Is the spectrum of base substitutions observed (A:T to G:C, A:T to C:G, A:T to T:A, etc.) different from a null hypothesis that these rates are uniform (that all six possible changes are equally likely)?</li>
</ol>
<div class="toggle container">
<div class="header container">
<strong>Show a hint</strong></div>
<div class="text container">
We need to compare two models, one with the same rate for all six types of base substitutions and one in which there are different rates allowed for each base substitution. This is a hard one. Here are some of the tools you&#8217;ll need to accomplish this in <strong>R</strong>:</div>
<div class="highlight-R"><div class="highlight"><pre><span class="o">&gt;</span> count <span class="o">=</span> read.csv<span class="p">(</span><span class="s">&quot;unique.csv&quot;</span><span class="p">,</span> header<span class="o">=</span><span class="bp">T</span><span class="p">)</span>
<span class="o">&gt;</span> <span class="kp">head</span><span class="p">(</span>count<span class="p">)</span>

<span class="c1"># Define the categories and columns with the numbers that we will be using</span>
<span class="o">&gt;</span> base.substitutions <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;AT.GC&quot;</span><span class="p">,</span> <span class="s">&quot;AT.CG&quot;</span><span class="p">,</span> <span class="s">&quot;AT.TA&quot;</span><span class="p">,</span> <span class="s">&quot;CG.TA&quot;</span><span class="p">,</span> <span class="s">&quot;CG.AT&quot;</span><span class="p">,</span> <span class="s">&quot;CG.GC&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> possible.columns <span class="o">=</span> <span class="kp">paste</span><span class="p">(</span><span class="s">&quot;POSSIBLE.TOTAL&quot;</span><span class="p">,</span> base.substitutions<span class="p">,</span>  sep<span class="o">=</span><span class="s">&quot;.&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> total.columns <span class="o">=</span> <span class="kp">paste</span><span class="p">(</span><span class="s">&quot;OBSERVED.TOTAL&quot;</span><span class="p">,</span> base.substitutions<span class="p">,</span>  sep<span class="o">=</span><span class="s">&quot;.&quot;</span><span class="p">)</span>

<span class="c1"># Create a data frame</span>
<span class="o">&gt;</span> base.sub <span class="o">=</span> <span class="kt">data.frame</span><span class="p">(</span>type <span class="o">=</span> <span class="kp">as.factor</span><span class="p">(</span>base.substitutions<span class="p">),</span> possible <span class="o">=</span> <span class="kp">as.numeric</span><span class="p">(</span>count<span class="p">[</span>possible.columns<span class="p">]),</span> total <span class="o">=</span> <span class="kp">as.numeric</span><span class="p">(</span>count<span class="p">[</span>total.columns<span class="p">]))</span>

<span class="c1"># Now we have the data organized like this:</span>
<span class="o">&gt;</span> base.sub
   type possible total
<span class="m">1</span> AT.GC  <span class="m">2279284</span>    <span class="m">14</span>
<span class="m">2</span> AT.CG  <span class="m">2279284</span>    <span class="m">23</span>
<span class="m">3</span> AT.TA  <span class="m">2279284</span>     <span class="m">9</span>
<span class="m">4</span> CG.TA  <span class="m">2350528</span>    <span class="m">37</span>
<span class="m">5</span> CG.AT  <span class="m">2350528</span>    <span class="m">14</span>
<span class="m">6</span> CG.GC  <span class="m">2350528</span>     <span class="m">3</span>

<span class="c1"># To fit the single rate model (with no intercept), use glm()</span>
<span class="o">&gt;</span> one.rate <span class="o">=</span> glm<span class="p">(</span>total <span class="o">~</span> possible <span class="o">+</span> <span class="m">0</span><span class="p">,</span> data<span class="o">=</span>base.sub<span class="p">,</span> family<span class="o">=</span>poisson<span class="p">())</span>
</pre></div>
</div>
</div>
<div class="toggle container">
<div class="header container">
<strong>Show the rest of the answer</strong></div>
<div class="highlight-R"><div class="highlight"><pre><span class="c1"># Fit the six rate model, use glm()</span>
<span class="o">&gt;</span> six.rate <span class="o">=</span> glm<span class="p">(</span>total <span class="o">~</span> possible <span class="o">+</span> type <span class="o">+</span> <span class="m">0</span><span class="p">,</span> data<span class="o">=</span>base.sub<span class="p">,</span> family<span class="o">=</span>poisson<span class="p">())</span>

<span class="c1"># Perform a likelihood ratio test to compare the model fits</span>
<span class="o">&gt;</span> anova<span class="p">(</span>one.rate<span class="p">,</span> six.rate<span class="p">,</span> test<span class="o">=</span><span class="s">&quot;LRT&quot;</span><span class="p">)</span>

Analysis of Deviance Table

Model <span class="m">1</span><span class="o">:</span> total <span class="o">~</span> possible <span class="o">+</span> <span class="m">0</span>
Model <span class="m">2</span><span class="o">:</span> total <span class="o">~</span> possible <span class="o">+</span> type <span class="o">+</span> <span class="m">0</span>
  Resid. Df Resid. Dev Df Deviance  Pr<span class="p">(</span><span class="o">&gt;</span>Chi<span class="p">)</span>
<span class="m">1</span>         <span class="m">5</span>     <span class="m">42.182</span>
<span class="m">2</span>         <span class="m">0</span>      <span class="m">0.000</span>  <span class="m">5</span>   <span class="m">42.182</span> <span class="m">5.411e-08</span> <span class="o">***</span>
<span class="o">---</span>
Signif. codes<span class="o">:</span>  <span class="m">0</span> ‘<span class="o">***</span>’ <span class="m">0.001</span> ‘<span class="o">**</span>’ <span class="m">0.01</span> ‘<span class="o">*</span>’ <span class="m">0.05</span> ‘<span class="m">.</span>’ <span class="m">0.1</span> ‘ ’ <span class="m">1</span>
</pre></div>
</div>
<div class="text container">
Therefore, the six rate model fits significantly better (P = 5.4E-8) , so we can reject the hypothesis that the rates of all six types of base pair substitutions are the same in this data set.</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">Be careful how you interpret these results! These are the rates at which mutations accumulate in the genomes <strong>after they are filtered by selection</strong>. They almost certainly do not represent the underlying rates at which new spontaneous mutations appear in the genomes of new cells in the population, which could be higher or lower.</p>
</div>
</div>
<ol class="arabic simple" start="2">
<li>Is there evidence of positive selection in the base substitutions that are observed in the evolving population (dN/dS &gt; 1)?</li>
</ol>
<div class="toggle container">
<div class="header container">
<strong>Show a hint</strong></div>
<div class="text container">
The observed dN/dS ratio can be calculated as:</div>
<div class="highlight-R"><div class="highlight"><pre><span class="o">&gt;</span> count <span class="o">=</span> read.csv<span class="p">(</span><span class="s">&quot;unique.csv&quot;</span><span class="p">,</span> header<span class="o">=</span><span class="bp">T</span><span class="p">)</span>

<span class="o">&gt;</span> dN <span class="o">=</span> count<span class="o">$</span>OBSERVED.NONSYNONYMOUS.TOTAL
<span class="o">&gt;</span> dS <span class="o">=</span> count<span class="o">$</span>OBSERVED.SYNONYMOUS.TOTAL
<span class="o">&gt;</span> dN<span class="o">/</span>dS
<span class="p">[</span><span class="m">1</span><span class="p">]</span> <span class="m">4.133333</span>
</pre></div>
</div>
<div class="text container">
However, you need to calculate the expected dN/dS ratio to see if this is elevated with respect to chance. This requires an assumption about the base change spectrum. Why? There is a much higher chance that a G:C to A:T mutation will be nonsynonymous than a G:C to T:A mutation, for example. So we need to sum over the number of possible synonymous and nonsynonymous mutations given a spectrum of base substitutions. For simplicity, let&#8217;s assume a 2:1 transition:transition ratio.</div>
</div>
<div class="toggle container">
<div class="header container">
<strong>Show the next hint</strong></div>
<div class="text container">
Here&#8217;s one way to calculate the expectation for dN/dS:</div>
<div class="highlight-R"><div class="highlight"><pre><span class="o">&gt;</span> base.substitutions <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;AT.GC&quot;</span><span class="p">,</span> <span class="s">&quot;AT.CG&quot;</span><span class="p">,</span> <span class="s">&quot;AT.TA&quot;</span><span class="p">,</span> <span class="s">&quot;CG.TA&quot;</span><span class="p">,</span> <span class="s">&quot;CG.AT&quot;</span><span class="p">,</span> <span class="s">&quot;CG.GC&quot;</span><span class="p">)</span>

<span class="c1"># This is a 2:1 transition to transversion ratio</span>
<span class="c1"># because there are two transversion categories for each transition</span>
<span class="o">&gt;</span> rate.spectrum <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">4</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">1</span><span class="p">)</span>

<span class="o">&gt;</span> possible.dN.columns <span class="o">=</span> <span class="kp">paste</span><span class="p">(</span><span class="s">&quot;POSSIBLE.NONSYNONYMOUS&quot;</span><span class="p">,</span> base.substitutions<span class="p">,</span>  sep<span class="o">=</span><span class="s">&quot;.&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> possible.dS.columns <span class="o">=</span> <span class="kp">paste</span><span class="p">(</span><span class="s">&quot;POSSIBLE.SYNONYMOUS&quot;</span><span class="p">,</span> base.substitutions<span class="p">,</span>  sep<span class="o">=</span><span class="s">&quot;.&quot;</span><span class="p">)</span>

<span class="o">&gt;</span> dN.expect <span class="o">=</span> <span class="kp">sum</span><span class="p">(</span>count<span class="p">[</span>possible.dN.columns<span class="p">]</span> <span class="o">*</span> rate.spectrum<span class="p">)</span>
<span class="o">&gt;</span> dS.expect <span class="o">=</span> <span class="kp">sum</span><span class="p">(</span>count<span class="p">[</span>possible.dS.columns<span class="p">]</span> <span class="o">*</span> rate.spectrum<span class="p">)</span>

<span class="o">&gt;</span> dN.expect<span class="o">/</span>dS.expect
<span class="p">[</span><span class="m">1</span><span class="p">]</span> <span class="m">2.320187</span>
</pre></div>
</div>
<div class="text container">
You still need to calculate whether the observed dN/dS ratio (4.13) is significantly greater than the expected ratio (2.32). To do this, use <strong>binom.test()</strong>.</div>
</div>
<div class="toggle container">
<div class="header container">
<strong>Show the answer</strong></div>
<div class="highlight-R"><div class="highlight"><pre><span class="o">&gt;</span> binom.test<span class="p">(</span><span class="kt">c</span><span class="p">(</span>dN<span class="p">,</span> dS<span class="p">),</span> p <span class="o">=</span> dN.expect<span class="o">/</span><span class="p">(</span>dN.expect <span class="o">+</span> dS.expect<span class="p">))</span>

Exact binomial test

data<span class="o">:</span>  <span class="kt">c</span><span class="p">(</span>dN<span class="p">,</span> dS<span class="p">)</span>
number of successes <span class="o">=</span> <span class="m">62</span><span class="p">,</span> number of trials <span class="o">=</span> <span class="m">77</span><span class="p">,</span> p<span class="o">-</span>value <span class="o">=</span> <span class="m">0.04629</span>
alternative hypothesis<span class="o">:</span> true probability of success is not equal to <span class="m">0.6988121</span>
<span class="m">95</span> percent confidence interval<span class="o">:</span>
 <span class="m">0.6991329</span> <span class="m">0.8866683</span>
sample estimates<span class="o">:</span>
probability of success
             <span class="m">0.8051948</span>
</pre></div>
</div>
<div class="text container">
<p>So the dN/dS ratio is marginally significantly different from expected with our assumed base mutation spectrum (P = 0.046).</p>
<p>For a (much) more conservative test, we can assume that the underlying base change spectrum is represented by the observed base changes (essentially a randomization test that keeps the base changes equal and shuffles them to random protein-coding positons in the genome). The code to do this is as follows:</p>
</div>
<div class="highlight-R"><div class="highlight"><pre><span class="o">&gt;</span> base.substitutions <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;AT.GC&quot;</span><span class="p">,</span> <span class="s">&quot;AT.CG&quot;</span><span class="p">,</span> <span class="s">&quot;AT.TA&quot;</span><span class="p">,</span> <span class="s">&quot;CG.TA&quot;</span><span class="p">,</span> <span class="s">&quot;CG.AT&quot;</span><span class="p">,</span> <span class="s">&quot;CG.GC&quot;</span><span class="p">)</span>

<span class="o">&gt;</span> possible.dN.columns <span class="o">=</span> <span class="kp">paste</span><span class="p">(</span><span class="s">&quot;POSSIBLE.NONSYNONYMOUS&quot;</span><span class="p">,</span> base.substitutions<span class="p">,</span>  sep<span class="o">=</span><span class="s">&quot;.&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> possible.dS.columns <span class="o">=</span> <span class="kp">paste</span><span class="p">(</span><span class="s">&quot;POSSIBLE.SYNONYMOUS&quot;</span><span class="p">,</span> base.substitutions<span class="p">,</span>  sep<span class="o">=</span><span class="s">&quot;.&quot;</span><span class="p">)</span>

<span class="o">&gt;</span> observed.dN.columns <span class="o">=</span> <span class="kp">paste</span><span class="p">(</span><span class="s">&quot;OBSERVED.NONSYNONYMOUS&quot;</span><span class="p">,</span> base.substitutions<span class="p">,</span>  sep<span class="o">=</span><span class="s">&quot;.&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> observed.dS.columns <span class="o">=</span> <span class="kp">paste</span><span class="p">(</span><span class="s">&quot;OBSERVED.SYNONYMOUS&quot;</span><span class="p">,</span> base.substitutions<span class="p">,</span>  sep<span class="o">=</span><span class="s">&quot;.&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> total.dN.dS <span class="o">=</span> count<span class="p">[</span>observed.dN.columns<span class="p">]</span> <span class="o">+</span> count<span class="p">[</span>observed.dS.columns<span class="p">]</span>
<span class="o">&gt;</span> total.dN.dS
  OBSERVED.NONSYNONYMOUS.AT.GC OBSERVED.NONSYNONYMOUS.AT.CG
<span class="m">1</span>                           <span class="m">10</span>                           <span class="m">19</span>
  OBSERVED.NONSYNONYMOUS.AT.TA OBSERVED.NONSYNONYMOUS.CG.TA
<span class="m">1</span>                            <span class="m">7</span>                           <span class="m">27</span>
  OBSERVED.NONSYNONYMOUS.CG.AT OBSERVED.NONSYNONYMOUS.CG.GC
<span class="m">1</span>                           <span class="m">11</span>                            <span class="m">3</span>

<span class="o">&gt;</span> dN.expect <span class="o">=</span> <span class="kp">sum</span><span class="p">(</span>count<span class="p">[</span>possible.dN.columns<span class="p">]</span> <span class="o">*</span> total.dN.dS<span class="p">)</span>
<span class="o">&gt;</span> dS.expect <span class="o">=</span> <span class="kp">sum</span><span class="p">(</span>count<span class="p">[</span>possible.dS.columns<span class="p">]</span> <span class="o">*</span> total.dN.dS<span class="p">)</span>

<span class="o">&gt;</span> dN.expect<span class="o">/</span>dS.expect
<span class="p">[</span><span class="m">1</span><span class="p">]</span> <span class="m">2.676483</span>

<span class="o">&gt;</span> binom.test<span class="p">(</span><span class="kt">c</span><span class="p">(</span>dN<span class="p">,</span> dS<span class="p">),</span> p <span class="o">=</span> dN.expect<span class="o">/</span><span class="p">(</span>dN.expect <span class="o">+</span> dS.expect<span class="p">))</span>

   Exact binomial test

data<span class="o">:</span>  <span class="kt">c</span><span class="p">(</span>dN<span class="p">,</span> dS<span class="p">)</span>
number of successes <span class="o">=</span> <span class="m">62</span><span class="p">,</span> number of trials <span class="o">=</span> <span class="m">77</span><span class="p">,</span> p<span class="o">-</span>value <span class="o">=</span> <span class="m">0.1581</span>
alternative hypothesis<span class="o">:</span> true probability of success is not equal to <span class="m">0.7280009</span>
<span class="m">95</span> percent confidence interval<span class="o">:</span>
 <span class="m">0.6991329</span> <span class="m">0.8866683</span>
sample estimates<span class="o">:</span>
probability of success
             <span class="m">0.8051948</span>
</pre></div>
</div>
<div class="text container">
As you can see, the dN/dS ratio is not significantly different from expected by this test. This is a very stringent test because positive selection (favoring an elevated dN/dS ratio) should actually alter the observed base spectrum from the underlying mutational spectrum. It will tend to favor transversions over transitions, because they are more likely to change the encoded amino acid.</div>
</div>
</div>
<div class="section" id="example-4-build-a-phylogenetic-tree-of-population-ara-3">
<h3>Example 4. Build a phylogenetic tree of population Ara-3<a class="headerlink" href="#example-4-build-a-phylogenetic-tree-of-population-ara-3" title="Permalink to this headline">¶</a></h3>
<p>You can use <strong>gdtools COMPARE</strong> and the <a class="reference external" href="http://evolution.genetics.washington.edu/phylip.html">Phylip package</a> program <strong>dnapars</strong> to create a maximum parsimony phylogenetic tree <strong class="program">GenomeDiff</strong> files.</p>
<p>At first, let&#8217;s just use a subset of the genomes through 30,000 generations. Move the others out of the <code class="docutils literal"><span class="pre">curated_gd</span></code> folder. We&#8217;ll keep the ancestral genome with no mutations (REL606) to use as the outgroup in rooting this tree.</p>
<p>See if you can figure out how to make a Phylip format output file from <strong>gdtools COMPARE</strong>:</p>
<div class="highlight-bash"><div class="highlight"><pre>gdtools COMPARE -h
</pre></div>
</div>
<div class="toggle container">
<div class="header container">
<strong>Show example code</strong></div>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span><span class="nb">cd </span>curated_gd
<span class="nv">$ </span>gdtools COMPARE -f phylip -o ../infile -r ../REL606.gbk <span class="sb">`</span>ls *.gd<span class="sb">`</span>
</pre></div>
</div>
</div>
<p>Examine the output file from <strong>gdtools COMPARE</strong> in a text editor. It consists of <strong>phylip</strong> input shown as nucleotides with a column for every unique mutation observed in the population. For SNPs, the actual nucleotides in the respective genome are shown in the column. For other mutations, an &#8216;A&#8217; is used for  the ancestral state, and a &#8216;T&#8217; is used for the derived state (e.g., with IS element inserted). You may also notice &#8216;N&#8217; states, which can occur when a region with a mutation in one genome was deleted in another genome, making the the state of that allele ambiguous. This &#8216;N&#8217; state can also occur when there were not enough reads in a sequencing sample to call mutations in this region.</p>
<p>This file is suitable for use as an <strong>infile</strong> to the phylip <strong>dnapars</strong> command. You can run <strong>dnapars</strong> at the command-line using a locally installed version of Phylip or run it on the web <a class="reference external" href="http://mobyle.pasteur.fr/cgi-bin/portal.py#forms::dnapars">via Mobyle &#64;Pasteur</a>. Use the default <strong>dnapars</strong> program parameters.</p>
<p>Now you can visualize the <strong>outree</strong> file (which is in <a class="reference external" href="https://en.wikipedia.org/wiki/Newick_format">Newick format</a>) via many different tools. Here are a couple that enable online viewing within a web browser:</p>
<ul class="simple">
<li><a class="reference external" href="http://www.trex.uqam.ca/index.php?action=newick">Newick Viewer &#64; T-REX web server</a></li>
<li><a class="reference external" href="http://www.phylogeny.fr/one_task.cgi?task_type=treedyn">Tree Dyn &#64; Phylogeny.fr</a></li>
</ul>
<p>You&#8217;ll notice that the names in your tree are the file names truncated to 10 characters (a limitation of the file format Phylip uses). This isn&#8217;t terribly useful for the way we have named the files. Here&#8217;s a version with the full names substituted back in: <a class="reference external" href="http://barricklab.org/release/breseq_tutorial/Ara-3.tre">download Ara-3.tre</a></p>
<div class="figure">
<img alt="_images/Ara-3.tree.png" src="_images/Ara-3.tree.png" />
</div>
<p>As a further exercise, you can try constructing a tree using all of the genomes from this population.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If you have <strong>phylip</strong> installed at the command line, you can also experiment with using <strong>gdtools PHYLOGENY</strong> to directly create a tree with the full names of each sample included.</p>
</div>
<p>What if we wanted to find the mutations that happened on a particular branch in the phylogenetic tree? We might want to know candidate mutations that could be responsible for a change in the phenotype, for example.</p>
<p>To do this, we can use our phylogenetic tree as a guide for <strong class="program">gdtools</strong> set operations. For example, let&#8217;s find out what mutations occurred on the branch highlighted in the figure above in red.</p>
<p>Use these commands:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>gdtools SUBTRACT -h
<span class="nv">$ </span>gdtools INTERSECT -h
</pre></div>
</div>
<div class="toggle container">
<div class="header container">
<strong>Here&#8217;s an example of how</strong></div>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span><span class="nb">cd </span>curated_gd
<span class="nv">$ </span>gdtools INTERSECT -o ../intersection.gd Ara-3_30000gen_ZDB16.gd Ara-3_30000gen_ZDB357.gd
<span class="nv">$ </span>gdtools SUBTRACT -o ../branch.gd ../intersection.gd Ara-3_10000gen_ZDB429.gd
<span class="nv">$ </span><span class="nb">cd</span> ..
<span class="nv">$ </span>gdtools COMPARE -r REL606.gbk -o branch.html branch.gd
</pre></div>
</div>
<div class="text container">
You can open <strong>branch.html</strong> in a web browser to see an annotated list of the 10 mutations along this branch.</div>
</div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="index.html">
              <img class="logo" src="_static/breseq.png" alt="Logo"/>
            </a></p>
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Tutorial: Clonal Samples (Consensus Mode)</a><ul>
<li><a class="reference internal" href="#download-data-files">1. Download data files</a><ul>
<li><a class="reference internal" href="#reference-sequence">Reference sequence</a></li>
<li><a class="reference internal" href="#read-files">Read files</a></li>
</ul>
</li>
<li><a class="reference internal" href="#run-breseq">2. Run <strong class="program">breseq</strong></a></li>
<li><a class="reference internal" href="#open-breseq-output">3. Open <strong class="program">breseq</strong> output</a></li>
<li><a class="reference internal" href="#resolving-the-cit-mutation">4. Resolving the Cit+ mutation</a><ul>
<li><a class="reference internal" href="#a-rnk-citg-junction">A. <em>rnk-citG</em> junction</a></li>
<li><a class="reference internal" href="#b-zoomed-in-coverage">B. Zoomed-in coverage</a></li>
<li><a class="reference internal" href="#c-add-the-amplification-to-the-genomediff-file">C. Add the amplification to the <cite>GenomeDiff</cite> file</a></li>
</ul>
</li>
<li><a class="reference internal" href="#generating-a-mutated-reference-sequence">5. Generating a mutated reference sequence</a></li>
<li><a class="reference internal" href="#characterizing-genetic-diversity-and-genome-evolution">6. Characterizing genetic diversity and genome evolution</a><ul>
<li><a class="reference internal" href="#example-1-compare-mutations-in-different-genomes">Example 1. Compare mutations in different genomes</a></li>
<li><a class="reference internal" href="#example-2-analyze-rates-and-nature-of-genome-evolution">Example 2. Analyze rates and nature of genome evolution</a></li>
<li><a class="reference internal" href="#example-3-analyze-unique-mutations-for-evidence-of-bias-and-selection">Example 3. Analyze unique mutations for evidence of bias and selection</a></li>
<li><a class="reference internal" href="#example-4-build-a-phylogenetic-tree-of-population-ara-3">Example 4. Build a phylogenetic tree of population Ara-3</a></li>
</ul>
</li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="acknowledgments.html"
                        title="previous chapter">Acknowledgments</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="tutorial_populations.html"
                        title="next chapter">Tutorial: Population Samples (Polymorphism Mode)</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/tutorial_clones.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="tutorial_populations.html" title="Tutorial: Population Samples (Polymorphism Mode)"
             >next</a> |</li>
        <li class="right" >
          <a href="acknowledgments.html" title="Acknowledgments"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">breseq 0.33.2 documentation</a> &raquo;</li> 
      </ul>
    </div>
 <script type="text/javascript">
    $(document).ready(function() {
        $(".toggle > *").hide();
        $(".toggle .header").show();
        $(".toggle .header").click(function() {
            $(this).parent().children().not(".header").toggle(400);
            $(this).parent().children(".header").toggleClass("open");
        })
    });
</script>

  </body>
</html>