<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Tutorial: Ultra-rare variant detection using consensus reads and targeted sequencing &mdash; breseq 0.33.2 documentation</title>
    
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/custom.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.33.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="breseq 0.33.2 documentation" href="index.html" />
    <link rel="next" title="Frequently Asked Questions (FAQ)" href="faq.html" />
    <link rel="prev" title="Tutorial: Population Samples (Polymorphism Mode)" href="tutorial_populations.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="faq.html" title="Frequently Asked Questions (FAQ)"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="tutorial_populations.html" title="Tutorial: Population Samples (Polymorphism Mode)"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">breseq 0.33.2 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="tutorial-ultra-rare-variant-detection-using-consensus-reads-and-targeted-sequencing">
<span id="tutorial-barcoded-targeted"></span><h1>Tutorial: Ultra-rare variant detection using consensus reads and targeted sequencing<a class="headerlink" href="#tutorial-ultra-rare-variant-detection-using-consensus-reads-and-targeted-sequencing" title="Permalink to this headline">¶</a></h1>
<p>In this exercise, you will analyze targeted sequencing of the initial burst of genetic diversity in a short <em>E. coli</em> evolution experiment. This tutorial uses data prepared by a special library preparation technique that adds a &#8220;molecular index&#8221; to each initial DNA fragment. This enables one to sequence many amplification products from this initial read to achieve lower error rates. In addition pulldowns with biotinylated oligos were used to enrich for only certain genes in the E. coli genome to achieve deeper sequencing of regions that were expected to have beneficial mutations.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This tutorial was created for the EMBO Practical Course <a class="reference external" href="http://events.embo.org/15-htp-sequencing/">Measuring intra-species diversity using high-throughput sequencing</a> held 27–31 July 2015 in Oeras, Portugal.</p>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">If you encounter any <strong class="program">breseq</strong> or <strong class="program">gdtools</strong> errors or crashes in running this tutorial, please <a class="reference external" href="https://github.com/barricklab/breseq/issues">report issues on GitHub</a>.</p>
</div>
<div class="section" id="download-data-files">
<h2>1. Download data files<a class="headerlink" href="#download-data-files" title="Permalink to this headline">¶</a></h2>
<p>First, create a directory called <strong>sscs_targeted</strong>:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>mkdir tutorial_barcoded_targeted
<span class="nv">$ </span><span class="nb">cd </span>tutorial_barcoded_targeted
</pre></div>
</div>
<div class="section" id="reference-sequence">
<h3>Reference sequence<a class="headerlink" href="#reference-sequence" title="Permalink to this headline">¶</a></h3>
<p>The samples sequenced were genomic DNA from populations evolved from a clonal isolate of <em>Escherichia coli</em> B strain REL606. We&#8217;ll contrast some new ways of analyzing this data that require us to use different reference genome setups.</p>
<p>First, download the entire reference genome.</p>
<p><a class="reference external" href="http://barricklab.org/release/breseq_tutorial/REL606.gbk.gz">Download REL606.gbk via this link</a></p>
<p>Then, download these two reference files:</p>
<ul class="simple">
<li><a class="reference external" href="http://barricklab.org/release/breseq_tutorial/on-target.gff3.gz">on-target.gff3</a></li>
<li><a class="reference external" href="http://barricklab.org/release/breseq_tutorial/off-target.gff3.gz">off-target.gff3</a></li>
</ul>
<p>In the first of these (<code class="docutils literal"><span class="pre">on-target.gff3</span></code>), we&#8217;ve extracted just the 8 target genes with 1400 bp added on each side as three different reference fragments. Each of these reference sequences were created by using <code class="docutils literal"><span class="pre">gdtools</span> <span class="pre">APPLY</span></code> to delete the rest of the genome.</p>
<p>In the second of these (<code class="docutils literal"><span class="pre">off-target.gff3</span></code>), we&#8217;ve masked (via N&#8217;s) the targeted regions in the whole genome sequence. This reference sequence was created by using <code class="docutils literal"><span class="pre">gdtools</span> <span class="pre">MASK</span></code>.</p>
<p>This separation of the reference sequence and the 8 targeted regions into different files and (artificial) DNA fragments will makes it easier to calculate certain statistics about enrichment of the target region relative to the rest of the genome and to ignore mutations that occur outside the regions of interest</p>
</div>
<div class="section" id="read-files">
<h3>Read files<a class="headerlink" href="#read-files" title="Permalink to this headline">¶</a></h3>
<p>Paired-end Illumina reads for one population sample taken at generation 139 from a time-course:</p>
<ul class="simple">
<li><a class="reference external" href="http://barricklab.org/release/breseq_tutorial/DED234_GATCAG_L004_R1_001.fastq.gz">DED234_GATCAG_L004_R1_001.fastq</a></li>
<li><a class="reference external" href="http://barricklab.org/release/breseq_tutorial/DED234_GATCAG_L004_R2_001.fastq">DED234_GATCAG_L004_R2_001.fastq</a></li>
</ul>
</div>
</div>
<div class="section" id="generate-sscs-reads">
<h2>2. Generate SSCS Reads<a class="headerlink" href="#generate-sscs-reads" title="Permalink to this headline">¶</a></h2>
<p>First, we need to pre-process the reads to construct single-strand consensus reads and remove the molecular barcodes. If you have numpy and other Python prerequisites installed, you can do this by downloading this script:</p>
<ul class="simple">
<li><a class="reference external" href="http://barricklab.org/release/breseq_tutorial/SSCS_DCS.py.gz">Download SSCS_DCS.py</a></li>
</ul>
<p>And then running this command:</p>
<div class="highlight-bash"><div class="highlight"><pre>python SSCS_DCS.py -f1 DED234_GATCAG_L004_R1_001.fastq -f2 DED234_GATCAG_L004_R2_001.fastq -p DED234 -s -d -m <span class="m">2</span> --log SSCS_Log
</pre></div>
</div>
<p>This script will take about 30 minutes to run.</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">This script is memory intensive! (16 GB RAM required)</p>
</div>
</div>
<div class="section" id="run-breseq-on-the-consensus-reads">
<h2>3. Run <strong class="program">breseq</strong> on the consensus reads<a class="headerlink" href="#run-breseq-on-the-consensus-reads" title="Permalink to this headline">¶</a></h2>
<p>Use this command to analyze the consensus reads.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>breseq -t -j <span class="m">8</span> -o consensus_reads -p --polymorphism-minimum-coverage-each-strand <span class="m">0</span> --polymorphism-bias-cutoff <span class="m">0</span> --polymorphism-score-cutoff <span class="m">0</span> --polymorphism-reject-indel-homopolymer-length <span class="m">0</span> --polymorphism-reject-surrounding-homopolymer-length <span class="m">0</span> -r on-target.gff3 -s off-target.gff3 DED234_SSCS.fastq
</pre></div>
</div>
<p>Notice the new <strong>-t</strong> and <strong>-s</strong> options. What are these doing? Look in the help under <strong>Reference File Options</strong>.</p>
<p>Take a look at the <strong class="program">breseq</strong> output. In particular, examine the <strong>summary statistics</strong> to look at the distribution of coverage across the different reference sequence fragments that were used. How effective was the enrichment step at favoring reads from those regions?</p>
</div>
<div class="section" id="run-breseq-on-the-original-reads">
<h2>4. Run <strong class="program">breseq</strong> on the original reads<a class="headerlink" href="#run-breseq-on-the-original-reads" title="Permalink to this headline">¶</a></h2>
<p>Your next task is to compare the performance of the consensus reads to the original reads. To make this comparison fair, you need to include the same numbers of original reads and consensus reads. Do this by extracting the same number of lines from the R1 file as exist in the new <code class="docutils literal"><span class="pre">DED234_SSCS.fastq</span></code> file.</p>
<div class="toggle container">
<div class="header container">
<strong>How? I need a hint</strong></div>
<div class="text container">
Check out the <strong>wc -l</strong> and <strong>head</strong> unix commands.</div>
</div>
<p>You also need to trim the molecular barcodes from these reads (this was done automatically by the <code class="docutils literal"><span class="pre">SSCS_DCS.py</span></code> script for us before).</p>
<p>Then, run a similar <strong class="program">breseq</strong> command to the one above to generate an <code class="docutils literal"><span class="pre">original_reads</span></code> output directory.</p>
<p>Compare the overall <strong class="program">breseq</strong> predictions on each data set (possibly by making a comparison table, as we have in the previous tutorials).</p>
<p>Also, take a look at a part of the <strong class="program">breseq</strong> output that you may not have examined yet. On the <strong>summary statistics</strong> page, click on the link named <strong>errors</strong> in the <strong>Read File Information</strong> table at the top. This graph shows the sequencing error rates in the input reads. Compare the results for the consensus and original reads.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="index.html">
              <img class="logo" src="_static/breseq.png" alt="Logo"/>
            </a></p>
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Tutorial: Ultra-rare variant detection using consensus reads and targeted sequencing</a><ul>
<li><a class="reference internal" href="#download-data-files">1. Download data files</a><ul>
<li><a class="reference internal" href="#reference-sequence">Reference sequence</a></li>
<li><a class="reference internal" href="#read-files">Read files</a></li>
</ul>
</li>
<li><a class="reference internal" href="#generate-sscs-reads">2. Generate SSCS Reads</a></li>
<li><a class="reference internal" href="#run-breseq-on-the-consensus-reads">3. Run <strong class="program">breseq</strong> on the consensus reads</a></li>
<li><a class="reference internal" href="#run-breseq-on-the-original-reads">4. Run <strong class="program">breseq</strong> on the original reads</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="tutorial_populations.html"
                        title="previous chapter">Tutorial: Population Samples (Polymorphism Mode)</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="faq.html"
                        title="next chapter">Frequently Asked Questions (FAQ)</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/tutorial_barcoded_targeted.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="faq.html" title="Frequently Asked Questions (FAQ)"
             >next</a> |</li>
        <li class="right" >
          <a href="tutorial_populations.html" title="Tutorial: Population Samples (Polymorphism Mode)"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">breseq 0.33.2 documentation</a> &raquo;</li> 
      </ul>
    </div>
 <script type="text/javascript">
    $(document).ready(function() {
        $(".toggle > *").hide();
        $(".toggle .header").show();
        $(".toggle .header").click(function() {
            $(this).parent().children().not(".header").toggle(400);
            $(this).parent().children(".header").toggleClass("open");
        })
    });
</script>

  </body>
</html>